executionlayer/src/main/kotlin/maru/executionlayer/client/PragueWeb3JJsonRpcExecutionLayerEngineApiClient.kt

```diff
@@ -32,28 +35,66 @@ import tech.pegasys.teku.infrastructure.bytes.Bytes8
 
 class PragueWeb3JJsonRpcExecutionLayerEngineApiClient(
   private val web3jEngineClient: Web3JExecutionEngineClient,
+  private val timerProvider: TimerProvider,
+  private val counterProvider: CounterProvider,
 ) : ExecutionLayerEngineApiClient {
   override fun getPayload(payloadId: Bytes8): SafeFuture<Response<ExecutionPayload>> =
-    web3jEngineClient.getPayloadV4(payloadId).thenApply {
-      when {
-        it.payload != null ->
-          Response.fromPayloadReceivedAsJson(it.payload.executionPayload.toDomainExecutionPayload())
+    timerProvider
+      .withTags(listOf(Tag("method", "getPayload")))
+      .captureTime(web3jEngineClient.getPayloadV4(payloadId))
+      .thenApply {
+        when {
+          it.payload != null -> {
+            counterProvider
```

Comment:
You don't really need counters as a separate thing. Counts are captured by timers themselves